src/main/java/org/opensearch/neuralsearch/highlight/NeuralHighlighter.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.highlight;
+
+import lombok.extern.log4j.Log4j2;
+import org.apache.lucene.search.Query;
+import org.opensearch.OpenSearchException;
+import org.opensearch.index.mapper.MappedFieldType;
+import org.opensearch.neuralsearch.ml.MLCommonsClientAccessor;
+import org.opensearch.neuralsearch.query.NeuralKNNQuery;
+import org.opensearch.search.fetch.subphase.highlight.FieldHighlightContext;
+import org.opensearch.search.fetch.subphase.highlight.HighlightField;
+import org.opensearch.search.fetch.subphase.highlight.Highlighter;
+import org.opensearch.core.common.text.Text;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Neural highlighter that uses ML models to identify relevant text spans for highlighting
+ */
+@Log4j2
+public class NeuralHighlighter implements Highlighter {
+    public static final String NAME = "neural";
+    private static final String MODEL_ID_FIELD = "model_id";
+
+    private static MLCommonsClientAccessor mlCommonsClient;
+
+    public static void initialize(MLCommonsClientAccessor mlClient) {
```

Comment:
This pattern is not good for unit test. Can we receive mlClient through constructor?